homeassistant/components/august/lock.py

```diff
@@ -43,27 +42,49 @@ def __init__(self, data, lock):
 
     async def async_lock(self, **kwargs):
         """Lock the device."""
-        update_start_time_utc = dt.utcnow()
-        lock_status = await self.hass.async_add_executor_job(
-            self._data.lock, self._lock.device_id
-        )
-        self._update_lock_status(lock_status, update_start_time_utc)
+        await self._call_lock_operation(self._data.lock)
 
     async def async_unlock(self, **kwargs):
         """Unlock the device."""
-        update_start_time_utc = dt.utcnow()
-        lock_status = await self.hass.async_add_executor_job(
-            self._data.unlock, self._lock.device_id
+        await self._call_lock_operation(self._data.unlock)
+
+    async def _linked_doorsense_entity_id(self):
+        if self._data.lock_has_doorsense(self._lock.device_id):
+            entity_registry = (
```

Comment:
Entities should not directly interact with the entity registry.

Instead, either use dispatcher and send signals or create an object where both the binary sensor and the lock can register so they can exchange messages or share data.

Reply:
This is something where you can decide to create a data update coordinator that a lock and binary sensor can share, so that they can trigger updates when it's available.

Reply:
We can't assume the entity exists, as individual entities can be disabled by the user in the entity registry.

Reply:
homeassistant.helpers.dispatcher looks that it will work for the activity updates as a more general solution.  Thanks. 

That's going to be a larger change so I'll take this out for now and solve it in another PR.

Reply:
Reverted for now. Added a note in the test that dispatcher needs to be implemented to fix the the existing issue (its not a new problem)